bridge/Bridge.cpp

```diff
@@ -91,6 +92,11 @@ Bridge::~Bridge()
 }
 
 void Bridge::initialize()
+{
+    initialize(std::shared_ptr<transport::EndpointFactory>(new transport::EndpointFactoryImpl()));
```

Comment:
We should use allows `make_shared`. It's more efficient as it use a single heap allocation to `EndpointFactoryImp` and ref counter controller, keep that memory close to each other and in C++14 `std:shared` + `new` as an argument of a function is not exception safe and it can create memory leaks if an exception is thrown.


```suggestion
    initialize(std::make_shared<EndpointFactoryImp>());
```

Reply:
Agree, fixed.